src/main/java/me/jellysquid/mods/lithium/common/entity/LithiumEntityCollisions.java

```diff
@@ -122,39 +122,25 @@ public boolean computeNext(Consumer<? super VoxelShape> consumer) {
                         continue;
                     }
 
-                    if (entity != null && entity.isConnectedThroughVehicle(otherEntity)) {
-                        continue;
-                    }
-
-                    Box otherEntityBox = otherEntity.getCollisionBox();
-
-                    boolean produced = false;
-
-                    if (otherEntityBox != null && box.intersects(otherEntityBox)) {
-                        if (consumer == null) {
-                            return true;
-                        } else {
-                            produced = true;
-                            consumer.accept(VoxelShapes.cuboid(otherEntityBox));
-                        }
-                    }
-
-                    if (entity != null) {
-                        Box otherEntityHardBox = entity.getHardCollisionBox(otherEntity);
-
-                        if (otherEntityHardBox != null && box.intersects(otherEntityHardBox)) {
-                            if (consumer == null) {
-                                return true;
-                            } else {
-                                produced = true;
-                                consumer.accept(VoxelShapes.cuboid(otherEntityHardBox));
-                            }
+                    /**
+                     * method_30948 returns false by default, designed to be overridden by entities whose collisions
+                     * should be "hard" (boats and shulkers, for now).
+                     * 
+                     * method_30949 only allows hard collisions if the calling entity is not riding otherEntity as a
+                     * vehicle.
+                     */
```

Comment:
These methods should be linked for easy access and remapping (when they get a mapping).
```suggestion
                    /**
                     * {@link Entity#method_30948} returns false by default, designed to be overridden by
                     * entities whose collisions should be "hard" (boats and shulkers, for now).
                     * 
                     * {@link Entity#method_30949} only allows hard collisions if the calling entity is not riding
                     * otherEntity as a vehicle.
                     */
```

Reply:
I'm trying to scope this PR as a pure port and avoid making any changes to Lithium proper. Another change I'd make is move then end noise caching optimization to the gen module instead of the world one, but that's probably best in another PR as well. Thank you for taking the time to test!